packages/node_modules/@ciscospark/internal-plugin-metrics/test/unit/spec/metrics.js

```diff
@@ -172,5 +173,150 @@ describe(`plugin-metrics`, () => {
+      });
+      it(`does not update on null args`, () => {
+        metrics._registerMetric(`horse`, `1`, testPayload, `clientMetric`);
+        const metric = metrics.getMetric(`1`);
+        assert.isFalse(metrics.updateRegisteredMetric(`1`, null));
+        assert.deepEqual(metric, metrics.getMetric(`1`));
+        assert.isFalse(metrics.updateRegisteredMetric(`2`, null));
+        // test null type branch
+        metrics._registerMetric(`testNullType`, `3`, testPayload, null);
+        assert.isFalse(metrics.updateRegisteredMetric(`3`, `someValue`));
+      });
+    });
+
+    describe(`#sendClientRegisteredMetric`, () => {
+      it(`sends client metric`, () => {
+        metrics._registerMetric(`horse`, `1`, testPayload, `clientMetric`);
+        return metrics.sendClientRegisteredMetric(`1`)
+          .then(() => {
+            assert.calledOnce(spark.request);
+            const req = spark.request.args[0][0];
+            const metric = req.body.metrics[0];
+
+            assert.property(metric, `metricName`);
+            assert.property(metric, `tags`);
+            assert.property(metric, `fields`);
+
+            assert.equal(metric.metricName, `horse`);
+            assert.equal(metric.tags.success, true);
+            assert.equal(metric.fields.perceivedDurationInMillis, testPayload.fields.perceivedDurationInMillis);
+          });
+      });
+      it(`does not send non-client metric`, () => {
+        metrics._registerMetric(`test`, `2`, testPayload, `sdkMetric`);
+        assert.isFalse(metrics.sendClientRegisteredMetric(`2`));
+      });
+      it(`does not send null metric`, () => {
+        assert.isFalse(metrics.sendClientRegisteredMetric(`4`));
+      });
+    });
+
+    describe(`#sendRegisteredMetric`, () => {
+      it(`sends sdk metric`, () => {
+        metrics._registerMetric(`horse`, `1`, testPayload, `sdkMetric`);
+        return metrics.sendRegisteredMetric(`1`)
+          .then(() => {
+            assert.calledOnce(spark.request);
+            const req = spark.request.args[0][0];
+            const metric = req.body.metrics[0];
+
+            assert.property(metric, `metricName`);
+            assert.property(metric, `tags`);
+            assert.property(metric, `fields`);
+
+            assert.equal(metric.metricName, `horse`);
+            assert.equal(metric.tags.success, true);
+            assert.equal(metric.fields.perceivedDurationInMillis, testPayload.fields.perceivedDurationInMillis);
+          });
+      });
+      it(`does not send non-sdj metric`, () => {
+        metrics._registerMetric(`test`, `2`, testPayload, `clientMetric`);
+        assert.isFalse(metrics.sendRegisteredMetric(`2`));
+      });
+      it(`does not send null metric`, () => {
+        assert.isFalse(metrics.sendRegisteredMetric(`4`));
+      });
+    });
+
+    describe(`#sendUnstructured`, () => {
+      it(`sends an unstructured metric`, () => {
+        return metrics.sendUnstructured(`foo`, `bar`)
```

Comment:
remove the return, move `metrics...` up a line, and use the arrow function's implicit return.

Reply:
done